Parse/Infrastructure/CacheController.cs

```diff
@@ -10,253 +10,304 @@
 
-            public bool ContainsKey(string key) => Lock(() => Storage.ContainsKey(key));
+        public bool ContainsKey(string key)
+        {
+            return Lock(() => Storage.ContainsKey(key));
+        }
 
-            public bool TryGetValue(string key, out object value)
+        public bool TryGetValue(string key, out object value)
+        {
+            lock (Mutex)
             {
-                lock (Mutex)
-                {
-                    return (Result: Storage.TryGetValue(key, out object found), value = found).Result;
-                }
+                return Storage.TryGetValue(key, out value);
             }
+        }
 
-            public void Clear() => Lock(() => Storage.Clear());
+        public void Clear()
+        {
+            Lock(() => Storage.Clear());
+        }
 
-            public bool Contains(KeyValuePair<string, object> item) => Lock(() => Elements.Contains(item));
+        public bool Contains(KeyValuePair<string, object> item)
+        {
+            return Lock(() => Elements.Contains(item));
+        }
 
-            public void CopyTo(KeyValuePair<string, object>[] array, int arrayIndex) => Lock(() => Elements.CopyTo(array, arrayIndex));
+        public void CopyTo(KeyValuePair<string, object>[] array, int arrayIndex)
+        {
+            Lock(() => Elements.CopyTo(array, arrayIndex));
+        }
 
-            public IEnumerator<KeyValuePair<string, object>> GetEnumerator() => Storage.GetEnumerator();
+        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+        {
+            return Storage.GetEnumerator();
+        }
 
-            IEnumerator IEnumerable.GetEnumerator() => Storage.GetEnumerator();
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return Storage.GetEnumerator();
+        }
 
-            public FileInfo File { get; set; }
+        public FileInfo File { get; set; }
 
-            public object Mutex { get; set; } = new object { };
+        public object Mutex { get; set; } = new object();
 
-            // ALTNAME: Operate
+        private TResult Lock<TResult>(Func<TResult> operation)
+        {
+            lock (Mutex)
+            {
+                return operation.Invoke();
+            }
+        }
 
-            TResult Lock<TResult>(Func<TResult> operation)
+        private void Lock(Action operation)
+        {
+            lock (Mutex)
             {
-                lock (Mutex)
-                {
-                    return operation.Invoke();
-                }
+                operation.Invoke();
             }
+        }
 
-            void Lock(Action operation)
+        private async Task LockAsync(Func<Task> operation)
+        {
+            lock (Mutex)
             {
-                lock (Mutex)
-                {
-                    operation.Invoke();
-                }
+                // Ensure operation runs within a locked context
```

Comment:
This does nothing right?

Reply:
I was supposed to ensure operations go on well, but that class's porting wasn't fully done from what I see, will readjust with more locks.
Just a moment (to test too)